### PR TITLE
DI configuration for the content type option provider

### PR DESCRIPTION
**Pull request by konarshankar07**
### Description (*)

This PR is introducing the DI configuration for the content type option provider for the `Magento\Framework\Data\OptionSourceInterface`.

### CODE REVIEW

**Review by sivaschenko — CHANGES_REQUESTED** [review]
Hi @konarshankar07 thanks for the pull request! As we discussed on the public meeting it's possible to use a class configuration similar to approach implemented for `\Magento\AdobeStockClient\Model\SearchParametersProvider\SimpleFilters`. Virtual type is not necessary in this case

**Comment by konarshankar07** [comment]
Hello @sivaschenko ...
Yes, I understood that but I still have a confusion but may be I'm wrong. If we use the real class then we still need to create separate class for each filter and repeating same code for each filter is not making sense to me. Correct me if I'm wrong
Thanks

**Comment by sivaschenko** [comment]
@konarshankar07 you will be able to extend arguments di.xml configuration for the real class the same way as for virtual class

File: AdobeStockImageAdminUi/Ui/Component/Listing/Columns/ContentType/Options.php [hunk 1]
```diff
@@ -11,26 +11,36 @@
...
+     */
+    public function __construct(array $options = [])
+    {
+        $this->_options = $options;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function toOptionArray()
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Ui/Component/Listing/Columns/ContentType/Options.php]:
```suggestion
    public function toOptionArray(): array
```

**Review by sivaschenko — DISMISSED** [review]

File: AdobeStockImageAdminUi/Ui/Component/Listing/Columns/ContentType/Options.php [hunk 2]
```diff
@@ -11,26 +11,36 @@
...
  */
 class Options implements OptionSourceInterface
 {
     /**
-     * Get options
-     *
-     * @return array
+     * @var array
+     */
+    protected $_options = [];
```
Comment (sivaschenko) [inline AdobeStockImageAdminUi/Ui/Component/Listing/Columns/ContentType/Options.php]:
```suggestion
    private $_options = [];
```

**Review by sivaschenko — APPROVED** [review]

**Pull request merged**